core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -333,7 +333,7 @@ private[spark] object PythonRDD extends Logging {
       valueConverterClass: String,
       confAsMap: java.util.HashMap[String, String],
       batchSize: Int): JavaRDD[Array[Byte]] = {
-    val conf = PythonHadoopUtil.mapToConf(confAsMap)
+    val conf = getMergedConf(confAsMap, sc.hadoopConfiguration())
```

Comment:
@advancedxy . We need a test case which failed without this PR. Please add a test case to show the validity of your PR.

Reply:
OK.  Will do.